include/dali/core/mm/default_resources.h

```diff
@@ -0,0 +1,98 @@
+// Copyright (c) 2021, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_CORE_MM_DEFAULT_RESOURCES_H_
+#define DALI_CORE_MM_DEFAULT_RESOURCES_H_
+
+
+#include <memory>
+#include "dali/core/api_helper.h"
+#include "dali/core/mm/memory_resource.h"
+
+namespace dali {
+namespace mm {
+
+template <memory_kind kind>
+struct DefaultMemoryResourceType;
+
+template <>
+struct DefaultMemoryResourceType<memory_kind::host> {
+  using type = host_memory_resource;
+};
+
+template <>
+struct DefaultMemoryResourceType<memory_kind::device> {
+  using type = device_async_resource;
+};
+
+template <>
+struct DefaultMemoryResourceType<memory_kind::pinned> {
+  using type = pinned_async_resource;
+};
+
+template <memory_kind kind>
+using default_memory_resource_t = typename DefaultMemoryResourceType<kind>::type;
+
+
+/**
+ * @brief Gets current memory resource for allocating memory of given kind.
+ */
+template <memory_kind kind>
+DLL_PUBLIC const std::shared_ptr<default_memory_resource_t<kind>> &GetDefaultResource();
+
+/**
+ * @brief Sets current memory resource for allocating memory of given kind.
```

Comment:
```suggestion
 * @brief Sets current memory resource for allocating memory of given kind (host or pinned).
```

Reply:
It works for device too (then it sets the default resource for the current device).

Reply:
ok